File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let set = new Set(nums);
        let max=0;

        for(let i=0;i< nums.length;i++){
            if(!set.has(nums[i]-1)){
                let length=0;
                while(set.has(nums[i]+length)){
                    length +=1;
                }
                max = Math.max(length,max)
            }
        }
        return max
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {

        int start = 0;
        int end = heights.length - 1;
        int max =0;

        while(start < end){
            int nums = (end - start) * Math.min(heights[start],  heights[end]);

            if(nums > max){
                max= nums;
            }

            if(heights[start] < heights[end]){
                start++;
            }else{
                end--;
            }
        }
        return max;
    }
}
